### ISSUE TITLE
Expose Swagger API documentation in official documentation

### ISSUE THREAD

**Issue by clems4ever** [authelia/authelia#2861]
### Description

It would be great to be able to review the API endpoints without deploying Authelia. It would allow to explore and discover features that are not sufficiently documented yet. For instance, this one was not: https://github.com/authelia/authelia/issues/2859 before it was raised.

**Comment by james-d-elliott** [comment]
Could potentially just add a link here: https://validator.swagger.io/?url=https://raw.githubusercontent.com/authelia/authelia/master/api/openapi.yml

**Comment by malsatin** [comment]
Hi.

The link, that you have provided now don't work.
It seems that the problem is in the added templating to openapi.yml, which happened in commit 241b9b19501c077ce43e0615042851e7b8b7ce87

**Comment by james-d-elliott** [comment]
Yeah the link unfortunately wont work anymore due to us templating the document live.

**Comment by polarathene** [comment]
Since Authelia is using Hugo for docs, perhaps this is helpful?: https://hugomods.com/docs/swagger-ui/

Looks like that would allow you to render the OpenAPI docs to embed static Swagger page?

I had to ask on the Discord community to know if Authelia documented the API anywhere that was easy to get an overview and browse. I was then told about the `/api` route providing Swagger docs at a running instance of Authelia (_`/docs/api` might have been more appropriate though? I assume the rest of `/api` subpaths are all headless_).

Is there a reason that the online docs for Authelia don't seem to mention this resource is available? Search for "swagger" and "openapi" did not return anything, nor didn't anything stand out from the many "api" / "/api" keywords.

If you do add this to the online docs with Hugo, it might be well suited as a tab-level item "API", much easier to discover than manually navigating through various docs pages or attempting to use the search feature. You should still document somewhere the runtime docs route too (and if this can be toggled?).

**Comment by james-d-elliott** [comment]
> Since Authelia is using Hugo for docs, perhaps this is helpful?: https://hugomods.com/docs/swagger-ui/

Yeah it's likely how I'd implement it if it became a priority.

> (/docs/api might have been more appropriate though? I assume the rest of /api subpaths are all headless).

The `/api` or `/api/v#` paths seem like a fairly common practice, regardless this is highly subjective.

> Is there a reason that the online docs for Authelia don't seem to mention this resource is available? Search for "swagger" and "openapi" did not return anything, nor didn't anything stand out from the many "api" / "/api" keywords.

Looks like it just got missed. Would welcome something in the reference or integration sections.

> If you do add this to the online docs with Hugo, it might be well suited as a tab-level item "API", much easier to discover than manually navigating through various docs pages or attempting to use the search feature. You should still document somewhere the runtime docs route too (and if this can be toggled?).

It's very unlikely we'll add another top level menu item, it belongs in the reference or integration sections as it really isn't a common request.
